### DISCUSSION TITLE
SSL Configuration for 2nd domain/server

### DISCUSSION THREAD

**Discussion by alex-pommier** [postalserver/postal#2105] [Help with using Postal]
I setup postal through docker on ubuntu and everything deployed fine and the emails are sending fine with click tracking through myprimarydomain.com 

I've now added a new set of IP's to be used on a second server in postal dedicated to mysecondarydomain.com for a different organization. I've setup the DNS and click tracking domains and I am able to send emails out fine however the click tracking doesn't work because the server is setup at myprimarydomain.com NOT mysecondarydomain.com so the rewriting fails SSL as there is none for mysecondarydomain.com. I've purchased a wildcard SSL to be used with click.mysecondarydomain.com I'd also like to access the postal interface via both domains so i've made updates to my config file like this:

##Postal.yml
web:
 host: postal.myprimarydomain.com **--existed already**
 host: postal.mysecondarydomain.com **--newly added**

##CaddyFile
postal.myprimarydomain.com { **--existed already**
  reverse_proxy 127.0.0.1:5000
}
postal.mysecondarydomain.com { **--newly added**
 reverse_proxy 127.0.0.1:5000
}

click.myprimarydomain.com { **--existed already**
 reverse_proxy 127.0.0.1:5000 {
  header_up X-Postal-Track-Host "1"
}
}
click.mysecondarydomain.com { **--newly added**
 reverse_proxy 127.0.0.1:5000 {
  header_up X-Postal-Track-Host "1"
}
}

However now anything I try visiting click.mysecondarydomain.com I'm getting cannot display issues due to couldn't establish a secure connection. I cannot find where the SSLs should be stored/added for my configuration. I purchased an SSL and can install it but don't know how or where the files should go. I didn't see any reference in postal.yml or CaddyFile for SSL certificates. Some guidance would be appreciated.

**Comment by willpower232** [comment]
> added a new set of IP's

a little off topic, have you set up IP pools for this? If you haven't then its probably not working how you think it is.

Adding a second `host` to postal.yml will do absolutely nothing, Postal is not made to be used like that.

That said, I think your caddy config should be sufficient for accessing the web UI but I'm not sure if Postal will just casually redirect you to the domain that it knows, I've experienced some weirdness in the past.

Caddy should automatically provide SSL certificates for the hosts so you don't need to obtain your own certificates unless you want to. If its not working then you'll have to check the caddy logs and documentation to see what is happening https://caddyserver.com/docs/automatic-https

**Reply by alex-pommier** [reply]
Hi Will,

Thank you for the info, after some time caddy did end up self-correcting the SSL issue and I was able to access it. The configuration did as you mentioned I was able to access the same postal server via 2 different domains but the sending is all happening from the primary as I described in my other discussion but I figured what I posted there is not possible I ended up just making a second postal server completely separate from my 1st and that works.
